### ISSUE TITLE
Terraform Crashed while planning azure keyvault module

### ISSUE THREAD

**Issue by ashwini19101994** [hashicorp/terraform#34997]
### Terraform Version

```shell
terraform {
  required_version = ">=v1.3.6"

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = ">3.50.0,<=3.58.0"
    }
  }
}
```

### Terraform Configuration Files

terraform {
  required_version = ">=v1.3.6"

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = ">3.50.0,<=3.58.0"
    }
  }
}

### Debug Output

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

panic: no expansion has been registered for module.key_vault
goroutine 302 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/1.22.1/x64/src/runtime/debug/stack.go:24 +0x5e
github.com/hashicorp/terraform/internal/logging.PanicHandler()
	/home/runner/work/terraform/terraform/internal/logging/panic.go:84 +0x1ac
panic({0x2a362a0?, 0xc0023cc5a0?})
	/opt/hostedtoolcache/go/1.22.1/x64/src/runtime/panic.go:770 +0x132
github.com/hashicorp/terraform/internal/instances.(*Expander).ExpandAbsModuleCall(0x2b70cc0?, {{0x0, 0x0, 0x0}, {{}, {0xc000653396, 0x9}}})
	/home/runner/work/terraform/terraform/internal/instances/expander.go:171 +0x265
github.com/hashicorp/terraform/internal/terraform.(*evaluationStateData).GetModule(0xc0004dcea0, {{}, {0xc000653396?, 0x0?}}, {{0xc000652a70, 0xa}, {0xd, 0x11, 0x12e}, {0xd, ...}})
	/home/runner/work/terraform/terraform/internal/terraform/evaluate.go:398 +0x788
github.com/hashicorp/terraform/internal/lang.(*Scope).evalContext(0xc0004dcf30, {0xc00152d478, 0x1, 0x1}, {0x0, 0x0})
	/home/runner/work/terraform/terraform/internal/lang/eval.go:381 +0x1112
github.com/hashicorp/terraform/internal/lang.(*Scope).EvalContext(...)
	/home/runner/work/terraform/terraform/internal/lang/eval.go:245
github.com/hashicorp/terraform/internal/lang.(*Scope).EvalExpr(0xc0004dcf30, {0x39b5e00, 0xc0009f1c80}, {{0x39b61e0?, 0x55175c0?}})
	/home/runner/work/terraform/terraform/internal/lang/eval.go:170 +0x9e
github.com/hashicorp/terraform/internal/terraform.(*BuiltinEvalContext).EvaluateExpr(0x0?, {0x39b5e00, 0xc0009f1c80}, {{0x39b61e0?, 0x55175c0?}}, {0x0?, 0x0?})
	/home/runner/work/terraform/terraform/internal/terraform/eval_context_builtin.go:322 +0xab
github.com/hashicorp/terraform/internal/terraform.(*NodeApplyableOutput).Execute(0xc0009e5980, {0x39d8ca8, 0xc000adaf00}, 0x70?)
	/home/runner/work/terraform/terraform/internal/terraform/node_output.go:376 +0x6cc
github.com/hashicorp/terraform/internal/terraform.(*ContextGraphWalker).Execute(0xc002c07e00, {0x39d8ca8, 0xc000adaf00}, {0x1a56a0bebd8, 0xc0009e5980})
	/home/runner/work/terraform/terraform/internal/terraform/graph_walk_context.go:153 +0xbb
github.com/hashicorp/terraform/internal/terraform.(*Graph).walk.func1({0x2edd6c0, 0xc0009e5980})
	/home/runner/work/terraform/terraform/internal/terraform/graph.go:112 +0x62a
github.com/hashicorp/terraform/internal/dag.(*Walker).walkVertex(0xc0009e5b00, {0x2edd6c0, 0xc0009e5980}, 0xc000844f00)
	/home/runner/work/terraform/terraform/internal/dag/walk.go:384 +0x2d7
created by github.com/hashicorp/terraform/internal/dag.(*Walker).Update in goroutine 374
	/home/runner/work/terraform/terraform/internal/dag/walk.go:307 +0xff3

### Expected Behavior

It should plan the resource and execute apply stage

### Actual Behavior

I am trying to deploy key vault module using azure devops pipelines. It was working fine in the morning but now getting this error.!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
Please report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version, the stack trace
shown below, and any additional information which may help replicate the issue.

[1]: https://github.com/hashicorp/terraform/issues

### Steps to Reproduce

terraform plan

**Comment by david-may-shift** [comment]
I am getting the same crash but at the terraform apply.

**Comment by liamcervante** [comment]
Thanks for filing this! Looking at the stack trace I think this is a duplicate of #34976, and will be fixed in v1.8.1.
